includes/REST_API/Link_Controller.php

```diff
@@ -83,6 +83,10 @@ public function register_routes() {
 	/**
 	 * Parses a URL to return some metadata for inserting links.
 	 *
+	 * @SuppressWarnings(PHPMD.CyclomaticComplexity)
```

Comment:
I agree with these lints. The method is too complex and long. Can we break it up anyway?

Reply:
Yes, in future PRs 🙂

Reply:
Worth documenting a ticket for this now?

Reply:
Maybe add a todo in the code to refactor.

Reply:
It's in the PR description at the moment.

Multiple follow-up tickets can be created once merged.

Reply:
See #5066 and its sub-tickets.